src/libexpr-c/nix_api_expr.h

```diff
@@ -14,6 +14,12 @@
 #include "nix_api_util.h"
 #include <stddef.h>
 
+#if _STDC_VERSION__ >= 202311L
```

Comment:
I think this can be done with a more precise feature check (untested):

```c
#if defined(__has_c_attribute) && __has_c_attribute(deprecated)
```

Reply:
Does not seem to work:

```
# foo.c
#if defined(__has_c_attribute) && __has_c_attribute(deprecated)
# error foo
#endif

int main() {
  return 0;
}
```

```
$ clang -std=c2x foo.c
```

This should error but it does not.

Reply:
```
% clang -std=c2x foo.c 
foo.c:2:3: error: foo
    2 | # error foo
      |   ^
1 error generated.
% clang --version                                
clang version 17.0.6
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /nix/store/klmyda94vsz4mkv1wnbp5v7cvfbz8qhc-clang-17.0.6/bin                                                                
```

Reply:
Got it working.